dali/kernels/alloc.cc

```diff
@@ -79,20 +83,62 @@ struct Allocator<AllocType::Unified> {
 };
 
 void *Allocate(AllocType type, size_t size) noexcept {
+#if !defined(__AARCH64_GNU__) && !defined(__AARCH64_QNX__)
   VALUE_SWITCH(type, type_label,
     (AllocType::Host, AllocType::Pinned, AllocType::GPU, AllocType::Unified),
     (return Allocator<type_label>::Allocate(size)),
     (assert(!"Invalid allocation type requested");
     return nullptr;)
   );  // NOLINT
+#else
+    void *ptr = nullptr;
```

Comment:
This macro was here exactly to avoid this kind of duplicated code.

Reply:
Yeah, i get that but QNX aarch64 toolchain does not identifies  VALUE_SWITCH  macro

Reply:
@mzient : Do you want me to remove the VALUE_SWITCH macro or continue with both of them? 
QNX toolchain does not identify such macro. Let me know your thoughts on this

Reply:
My thought is: _if the license allows this_, bundle boost/preprocessor with DALI. We have a third_party directory and we could put this part of boost there.

Reply:
I believe you just need to add it in Dockerfile.build.aarch64-linux (see my comment there)

Reply:
andi4919 said:
> Yes, boost was creating issues for QNX cross-compilation
In that case, pulling a (generally) unusable library just to use one module may be misleading to our users. On the other hand, shipping boost/preprocessor with DALI might create include directory ambiguity.

Reply:
Isn't boost for CPU based parallelization of algorithms? I am wondering why would it be useful for iGPUs?

Reply:
Boost is a general-purpose C++ utility library. It contains just about anything: mathematical functions, threading, parser generators, graph algorithms, type traits and more. I used preprocessor utilities from boost to avoid copy-pasting code. The preprocessor macro utlities are as usable on GPU as they are on CPU.

Reply:
@mzient: Removed the duplicated code. Using VALUE_SWITCH macro from boost  now